src/assembler/instruction.h

```diff
@@ -289,14 +291,51 @@ class Instruction {
     }
 
     AssembleRes assemble(const Assembler::TokenizedSrcLine& line) const {
+        QString Hint = "";
+        // if(dynamic_cast<Imm*>(m_fields[1].get()))
+        //     std::cout << "Its Imm" << std::endl;
+        // else if(dynamic_cast<Reg*>(m_fields[1].get()))
+        //     std::cout << "Its Reg" << std::endl;
+        // else
+        //     std::cout << "Nothing" << std::endl;
+
+        for(int i = 0; i < (m_expectedTokens-1); i++){
```

Comment:
The reason why you're having to do `i < (m_expectedTokens-1)` here is because `m_expectedtokens` counts the number of tokens excluding the opcode.
In any case, i'd suggest you to drop the old-style for loop and use something like this instead
`for(const auto& field : m_fields) { ...`